tax/totals.go

```diff
@@ -207,13 +210,12 @@ func (ct *CategoryTotal) calculate(zero num.Amount) {
 
 func (rt *RateTotal) matches(c *Combo) bool {
 	if c.Percent == nil {
-		// If there is no percent, try matching key
-		return c.Rate != cbc.KeyEmpty && c.Rate == rt.Key
+		return rt.Key == c.Rate && rt.Ext.Equals(c.Ext)
 	}
-	if rt.Key.IsEmpty() {
+	if (c.Rate.IsEmpty() || rt.Key == c.Rate) && rt.Ext.Equals(c.Ext) {
 		return rt.percentagesMatch(c)
 	}
-	if c.Rate.IsEmpty() || rt.Key == c.Rate {
+	if rt.Ext.Equals(c.Ext) {
 		return rt.percentagesMatch(c)
```

Comment:
Valid point, but my expectation here is that in practical terms invoices will not have more than one usage of the `ext`, its a "just in case" feature. My feeling is that we should wait until we have more concrete examples where grouping like this is a problem.

Reply:
You may be right that this won't become a real problem. Thing is, if it weren't because of the Italian case, it would have never crossed our minds to implement this. My feeling is that implementing this kind of unintuitive behaviours to solve individual cases adds accidental complexity that makes things increasingly harder to understand and change in the future than it could be. And if we wait to discover real cases where a behaviour like this is problematic, it may be too late (i.e. too expensive) to change it.